### PR TITLE
Unit test entites

### LINKED ISSUE #127: [B] Define execution entity structure
In order to show history, we need to find a way to store the result of an execution. It should include the server, database, *unit test conditions*, sample data, and results.

### PR DESCRIPTION
**Pull request by adityavidyadharan**
# Description

Defined the basic structure for the entities needed to save unit tests to memory

## Link to the issue you are closing

Fixes #127 

## Screenshot or other testing that you have completed

Screen Shot 2022-10-13 at 5 54 33 PM

## Test code
```ts
public async test() {
    await AppDataSource.initialize();
    setLog();
    const c1 = await new ConnectionService().create({
      id: 1,
      type: DBProvider.PostgreSQL,
      nickname: 'something_dumb',
      connectionConfig: {
        defaultDatabase: 'React',
        config: 'manual',
        username: 'kpmg',
        password: 'asdf',
        address: 'localhost',
        port: 5432,
      },
    });
    const c2 = await new ConnectionService().create({
      id: 2,
      type: DBProvider.PostgreSQL,
      nickname: 'something_dumber',
      connectionConfig: {
        defaultDatabase: 'React',
        config: 'manual',
        username: 'kpmg',
        password: 'asdf',
        address: 'localhost',
        port: 5432,
      },
    });

    await Promise.all(
      [1, 2, 3].map(async (id) => {
        const execution = new ExecutionEntity();
        execution.id = id;
        execution.timestamp = new Date();
        execution.testData = 'test';
        execution.connections = [c1, c2];

        return this.repository.save(execution);
      })
    );

    await Promise.all(
      [4, 5, 6].map(async (id) => {
        const execution = new ExecutionEntity();
        execution.id = id;
        execution.timestamp = new Date();
        execution.testData = 'test';
        execution.connections = [c2];

        return this.repository.save(execution);
      })
    );

    await new ConnectionService().repository.remove(c2);
  }
```

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

**Comment by adityavidyadharan** [comment]
Pending team discussion before merging

File: src/db/entity/UnitTestEntity.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  ManyToOne,
+  TableInheritance,
+  ChildEntity,
+} from 'typeorm';
+import { RowOperations, TableOperations } from './enum';
+import RuleEntity from './RuleEntity';
+
+@TableInheritance({ column: { type: 'varchar', name: 'type' } })
+@Entity({ name: 'UnitTest' })
+export class UnitTestEntity {
```
Comment (insaan21) [inline src/db/entity/UnitTestEntity.ts:14]:
```suggestion
class UnitTestEntity {
```

File: src/db/entity/ExecutionEntity.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  id: number;
+
+  @Column()
+  timestamp: Date;
+
+  /**
+   * A list of the rules being tested
+   */
+  @OneToMany((_type) => RuleEntity, (rule) => rule)
+  rules: RuleEntity[];
```
Comment (insaan21) [inline src/db/entity/ExecutionEntity.ts:27]: I think you'll need an on delete cascade here to delete all the rule groups once an execution is deleted?

File: src/db/entity/ExecutionEntity.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  /**
+   * The server connections to execute on
+   */
+  @ManyToMany(
+    (_type) => ConnectionEntity,
+    (connection) => connection.executions,
+    {
+      onDelete: 'CASCADE',
```
Comment (insaan21) [inline src/db/entity/ExecutionEntity.ts:36]: You also might want to add on update cascades everywhere: https://www.sqlite.org/foreignkeys.html#:~:text=The%20ON%20DELETE%20and%20ON,defaults%20to%20%22NO%20ACTION%22.
There should be some way to just easily configure this globally

**Review by insaan21 — DISMISSED** [review]
This looks good apart from the small things I mentioned. Actually super neat!

File: src/db/entity/ExecutionEntity.ts (hunk 3 above)
Comment (insaan21) [reply src/db/entity/ExecutionEntity.ts:36]: Never mind there isn't, might just have to on update cascades everywhere

File: src/db/entity/UnitTestEntity.ts (hunk 1 above)
Comment (adityavidyadharan) [reply src/db/entity/UnitTestEntity.ts:14]: Class is used in `RuleEntity`

**Review by insaan21 — DISMISSED** [review]
Make Connection to Execution Many to One

**Review by insaan21 — APPROVED** [review]

**Pull request merged**